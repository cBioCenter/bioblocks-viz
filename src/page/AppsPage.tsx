import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header, List } from 'semantic-ui-react';

import { IVizSummaryData, VizData } from '~chell-viz~/data';

export interface IAppsPageProps extends Partial<RouteComponentProps> {}

export class AppsPage extends React.Component<IAppsPageProps, any> {
  constructor(props: IAppsPageProps) {
    super(props);
  }

  public render() {
    return (
      <List divided={true}>
        <List.Item>{this.renderAppItem(VizData.spring)}</List.Item>
        <List.Item>{this.renderAppItem(VizData.tfjsTsne)}</List.Item>
        <List.Item>{this.renderAppItem(VizData.anatomogram)}</List.Item>
      </List>
    );
  }

  protected renderAppItem(viz: IVizSummaryData) {
    return (
      <Grid centered={true} columns={3} padded={true} relaxed={true}>
        <Grid.Column>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            alt={`icon for ${viz.name}`}
            style={{ height: '100px' }}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header as={'h2'}>{viz.name}</Header>
          <List>
            <List.Item>{`${viz.listAsOriginal ? 'original: ' : ''}${viz.authors.join(', ')}`}</List.Item>
            <List.Item>{viz.summary}</List.Item>
            <List.Item>{`relevant data: ${viz.relevantData}`}</List.Item>
          </List>
        </Grid.Column>
        <Grid.Column stretched={true}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/overview', search: `?name=${viz.name.toLocaleLowerCase()}` }}>details</Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={'/beta'}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid>
    );
  }
}
