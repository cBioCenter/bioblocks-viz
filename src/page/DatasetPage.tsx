import { fromJS, List } from 'immutable';
import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Grid, Message } from 'semantic-ui-react';

import { AnatomogramContainer, SpringContainer, TensorTContainer } from '~chell-viz~/container';
import { VizData } from '~chell-viz~/data';

export interface IDatasetPageProps extends Partial<RouteComponentProps> {}

export interface IDatasetPageState {
  visualizations: List<string>;
  datasetLocation: string;
}

export class DatasetPage extends React.Component<IDatasetPageProps, IDatasetPageState> {
  constructor(props: IDatasetPageProps) {
    super(props);
    this.state = {
      datasetLocation: '',
      visualizations: List<string>(),
    };
  }

  public componentDidMount() {
    if (this.props.location) {
      this.setupSearchParameters(this.props.location.search);
    }
  }

  public componentDidUpdate(prevProps: IDatasetPageProps) {
    if (this.props.location && this.props.location !== prevProps.location) {
      this.setupSearchParameters(this.props.location.search);
    }
  }

  public render() {
    const { visualizations, datasetLocation } = this.state;

    return (
      <div style={{ padding: '20px' }}>
        <Grid centered={true} stackable={true} stretched={false} padded={true} columns={2}>
          {datasetLocation.length >= 1 &&
            visualizations.map((visualization, index) => (
              <Grid.Column key={`dataset-visualization-${index}`} style={{ width: 'auto' }}>
                {this.renderVisualization(visualization, datasetLocation)}
              </Grid.Column>
            ))}
        </Grid>
      </div>
    );
  }

  protected setupSearchParameters(query: string) {
    const params = new URLSearchParams(query);
    // tslint:disable-next-line:no-backbone-get-set-outside-model
    const datasetLocation = params.get('name');
    const visualizations = fromJS(params.getAll('viz')) as List<string>;

    this.setState({
      datasetLocation: datasetLocation ? datasetLocation : '',
      visualizations,
    });
  }

  protected renderVisualization(viz: string | undefined, datasetLocation: string) {
    const isFullPage = this.state.visualizations.size === 1;
    switch (viz) {
      case VizData.spring.name.toLocaleLowerCase():
        return <SpringContainer datasetLocation={datasetLocation} isFullPage={isFullPage} />;
      case VizData.tfjsTsne.name.toLocaleLowerCase():
        return <TensorTContainer datasetLocation={datasetLocation} isFullPage={isFullPage} />;
      case VizData.anatomogram.name.toLocaleLowerCase():
        return <AnatomogramContainer species={datasetLocation.includes('hpc') ? 'homo_sapiens' : 'mus_musculus'} />;
      default:
        return <Message error={true}>{`Currently unsupported visualization '${viz}'`}</Message>;
    }
  }
}
