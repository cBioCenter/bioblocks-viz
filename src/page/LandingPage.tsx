import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Container, Divider, Grid, Header, Segment } from 'semantic-ui-react';

import { IDatasetInfo, IStory, IVizSummaryData, Stories, userDatasets, VizData } from '~chell-viz~/data';

export interface ILandingPageProps extends Partial<RouteComponentProps> {}

export class LandingPage extends React.Component<ILandingPageProps, any> {
  constructor(props: ILandingPageProps) {
    super(props);
  }

  public render() {
    return (
      <Container fluid={false}>
        <Segment basic={true} padded={'very'}>
          <Grid centered={true} padded={true} relaxed={true}>
            {this.renderFeaturedStories()}
            {this.renderFeaturedVisualizations()}
            {this.renderFeaturedDatasets()}
          </Grid>
        </Segment>
      </Container>
    );
  }

  protected renderSingleVisualization(viz: IVizSummaryData) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img
            src={`assets/icons/${viz.name.toLocaleLowerCase()}-thumbnail.png`}
            style={{ height: '90px', width: '90px' }}
            alt={`viz ${viz.name} icon`}
          />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{viz.name}</Header>
          {viz.summary}
          <p>
            <span style={{ fontWeight: 'bold' }}>{viz.listAsOriginal ? 'Original authors' : 'Authors'}: </span>
            {viz.authors.length === 2 ? `${viz.authors[0]} and ${viz.authors[1]}` : viz.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column width={2}>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/visualizations/', search: `?name=${viz.name.toLocaleLowerCase()}` }}>
                details
              </Link>
            </Button>
          </Grid.Row>
          <Grid.Row>
            <Button basic={true}>
              <Link to={{ pathname: '/dataset', search: `?viz=${viz.name.toLocaleLowerCase()}` }}>launch</Link>
            </Button>
          </Grid.Row>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderFeaturedDatasets() {
    return (
      <>
        <Grid.Row centered={true}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Datasets</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={2} divided={true}>
          {this.renderHCADatasets()}
          {this.renderUserSharedDatasets()}
        </Grid.Row>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'datasets'}>
              more datasets...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedStories() {
    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Stories</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {this.renderSingleFeaturedStory(Stories[0])}
        {this.renderSingleFeaturedStory(Stories[1])}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'stories'}>
              more stories...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderFeaturedVisualizations() {
    return (
      <>
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Header floated={'left'}>Featured Visualizations</Header>
            <Divider section={true} />
          </Grid.Column>
        </Grid.Row>
        {this.renderSingleVisualization(VizData.spring)}
        {this.renderSingleVisualization(VizData.tfjsTsne)}
        <Grid.Row centered={false}>
          <Grid.Column width={12}>
            <Link style={{ color: 'blue', float: 'right' }} to={'visualizations'}>
              more visualizations...
            </Link>
          </Grid.Column>
        </Grid.Row>
      </>
    );
  }

  protected renderSingleFeaturedStory(story: IStory) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column width={2}>
          <img src={story.icon} style={{ height: '90px', width: '90px' }} alt={`story ${story.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'} width={8}>
          <Header>{story.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {story.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {story.authors.length === 2 ? `${story.authors[0]} and ${story.authors[1]}` : story.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column width={2}>
          <Button basic={true}>
            <Link to={story.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderHCADatasets() {
    return (
      <Grid.Column width={3}>
        <Grid.Row>
          <img
            src={'assets/icons/hca-logo.png'}
            style={{ width: '150px', height: '250px' }}
            alt={'human cell atlas logo'}
          />
          <Grid.Row>
            <a href={'https://preview.data.humancellatlas.org/'}>Search HCA datasets ...</a>
          </Grid.Row>
        </Grid.Row>
      </Grid.Column>
    );
  }

  protected renderSingleUserSharedDatasets(dataset: IDatasetInfo) {
    return (
      <Grid.Row columns={2}>
        <Grid.Column floated={'left'}>
          <Header>{dataset.name}</Header>
          <p>
            {dataset.summary}
            <br />
            <span style={{ fontWeight: 'bold' }}>Authors: </span>
            {dataset.authors.join(', ')}
          </p>
        </Grid.Column>
        <Grid.Column floated={'right'}>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.detail}>details</Link>
          </Button>
          <Button basic={true} floated={'right'}>
            <Link to={dataset.links.analysis}>analyze</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }

  protected renderUserSharedDatasets() {
    return (
      <Grid.Column stretched={true} textAlign={'left'} width={8}>
        <Grid.Row textAlign={'center'}>
          <Header as={'h2'} textAlign={'center'}>
            User shared
          </Header>
        </Grid.Row>
        <br />
        <Grid.Row>{this.renderSingleUserSharedDatasets(userDatasets[0])}</Grid.Row>
        <Grid.Row>{this.renderSingleUserSharedDatasets(userDatasets[1])}</Grid.Row>
      </Grid.Column>
    );
  }
}
