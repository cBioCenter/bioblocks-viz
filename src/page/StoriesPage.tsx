import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';
import { Button, Grid, Header } from 'semantic-ui-react';

import { IStory, Stories } from '~chell-viz~/data';

export interface IStoriesPageProps extends Partial<RouteComponentProps> {}

export class StoriesPage extends React.Component<IStoriesPageProps, any> {
  constructor(props: IStoriesPageProps) {
    super(props);
  }

  public render() {
    return (
      <Grid centered={true} divided={'vertically'} padded={true} relaxed={false}>
        {Stories.map((story, index) => (
          <React.Fragment key={`story-${index}`}>{this.renderSingleFeaturedStory(story)}</React.Fragment>
        ))}
      </Grid>
    );
  }

  protected renderSingleFeaturedStory(story: IStory) {
    return (
      <Grid.Row columns={3}>
        <Grid.Column>
          <img src={story.icon} style={{ height: '90px', width: '90px' }} alt={`story ${story.title} icon`} />
        </Grid.Column>
        <Grid.Column textAlign={'left'}>
          <Header>{story.title}</Header>
          <p>
            <span style={{ fontWeight: 'bold' }}>Description: </span>
            {story.description}
          </p>
          <p>
            <span style={{ fontWeight: 'bold' }}>Analysis authors: </span>
            {story.authors.length === 2 ? `${story.authors[0]} and ${story.authors[1]}` : story.authors.join(', ')}
          </p>
          <br />
        </Grid.Column>
        <Grid.Column>
          <Button basic={true}>
            <Link to={story.link}>launch</Link>
          </Button>
        </Grid.Column>
      </Grid.Row>
    );
  }
}
