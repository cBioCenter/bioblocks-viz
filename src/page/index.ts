export * from './VisualizationsPage';
export * from './DatasetPage';
export * from './LandingPage';
export * from './StoriesPage';
export * from './VizOverviewPage';
