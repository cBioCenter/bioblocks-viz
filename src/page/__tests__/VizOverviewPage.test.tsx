import { shallow } from 'enzyme';
import * as React from 'react';

import { VizOverviewPage } from '~chell-viz~/page';

describe('VizOverviewPage', () => {
  const visualizations = ['anatomogram', 'spring', 'tfjs-tsne'];

  it('Should match existing snapshot when no props are provided.', () => {
    const wrapper = shallow(<VizOverviewPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot for initial visualizations.', () => {
    visualizations.forEach(viz => {
      describe(viz, () => {
        const wrapper = shallow(
          <VizOverviewPage location={{ hash: '', pathname: '', search: `?name=${viz}`, state: '' }} />,
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  it('Should match existing snapshot when changing the visualization.', () => {
    const wrapper = shallow(
      <VizOverviewPage location={{ hash: '', pathname: '', search: '?name=spring', state: '' }} />,
    );
    wrapper.setProps({
      location: { hash: '', pathname: '', search: '?name=anatomogram', state: '' },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
