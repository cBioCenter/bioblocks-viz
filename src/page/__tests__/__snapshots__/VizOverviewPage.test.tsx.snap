// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VizOverviewPage Should match existing snapshot for initial visualizations. 1`] = `
<Container>
  <Grid
    centered={true}
    columns={2}
  >
    <GridColumn
      width={2}
    >
      <img
        alt="icon for Anatomogram"
        src="assets/icons/anatomogram-icon.png"
        style={
          Object {
            "height": "150px",
            "padding": "20px",
          }
        }
      />
    </GridColumn>
    <GridColumn
      textAlign="left"
    >
      <Header
        as="h1"
      >
        Anatomogram
        <HeaderSubheader>
          Alfonso Mu√±oz-Pomer Fuentes
        </HeaderSubheader>
      </Header>
      <p />
      <List>
        <ListItem>
          applicable data: 
          scRNA-seq
        </ListItem>
        <ListItem>
          compatible with: 
        </ListItem>
        <ListItem>
          citation(s):
           
        </ListItem>
        <ListItem>
          version: 
           (last updated 
          ),
          <a
            href="https://github.com/ebi-gene-expression-group/anatomogram"
          >
             github link
          </a>
        </ListItem>
        <ListItem>
          <GridColumn
            floated="right"
          >
            <Button
              as="button"
              basic={true}
              icon={true}
              labelPosition="right"
              role="button"
            >
              <Link
                replace={false}
                to={
                  Object {
                    "pathname": "/dataset",
                    "search": "?viz=anatomogram",
                  }
                }
              >
                launch Anatomogram
              </Link>
              <Icon
                as="i"
                name="external alternate"
              />
            </Button>
          </GridColumn>
        </ListItem>
      </List>
    </GridColumn>
  </Grid>
  <Accordion
    defaultActiveIndex={0}
    panels={
      Array [
        Object {
          "content": Object {
            "content": <React.Fragment>
              <Divider />
              <Grid
                centered={true}
                columns={1}
                container={true}
                divided="vertically"
                style={
                  Object {
                    "padding": "10px 0",
                  }
                }
              >
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="Tabula Muris - SPRING vs tSNE with Anatomogram icon"
                        src="assets/icons/example_tabularmuris_spring-tsne-anatomogram.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        Tabula Muris - SPRING vs tSNE with Anatomogram
                      </Header>
                      <p>
                        Example comparison of data reduction techniques SPRING and tSNE    (of top 30 PCA components), on RNAseq profiles from 10000 cells    from the CZI Tabula muris dataset. Also includes anatomogram visualization for selection of tissue types.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=tabula_muris/10k&viz=spring&viz=tfjs-tsne&viz=anatomogram"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
              </Grid>
            </React.Fragment>,
          },
          "key": "examples",
          "title": "example stories",
        },
        Object {
          "content": "coming soon!",
          "key": "learn",
          "title": "learn",
        },
      ]
    }
  />
</Container>
`;

exports[`VizOverviewPage Should match existing snapshot for initial visualizations. 2`] = `
<Container>
  <Grid
    centered={true}
    columns={2}
  >
    <GridColumn
      width={2}
    >
      <img
        alt="icon for SPRING"
        src="assets/icons/spring-icon.png"
        style={
          Object {
            "height": "150px",
            "padding": "20px",
          }
        }
      />
    </GridColumn>
    <GridColumn
      textAlign="left"
    >
      <Header
        as="h1"
      >
        SPRING
        <HeaderSubheader>
          Caleb Weinreb, Samuel Wolock, Allon Klein
        </HeaderSubheader>
      </Header>
      <p>
        SPRING is a tool for uncovering high-dimensional structure in single-cell gene expression data.   SPRING takes a (gene X cell) table of expression measurements and outputs a k-nearest-neighbor graph   rendered using a force directed layout.   Users can overlay prior information, including gene expression values, gene-set scores, cell cluster labels and sample IDs.   Users can also upload custom coordinates that have been generated using an outside program such as tSNE.
      </p>
      <List>
        <ListItem>
          applicable data: 
          scRNA-seq
        </ListItem>
        <ListItem>
          compatible with: 
          live tSNE, UMAP, PCA
        </ListItem>
        <ListItem>
          citation(s):
           
          Weinreb, Caleb, Samuel Wolock, and Allon M. Klein.      "SPRING: A kinetic interface for visualizing high dimensional single-cell expression data."      Bioinformatics 34.7 (2017): 1246-1248.
           (
          <a
            href="https://www.ncbi.nlm.nih.gov/pubmed/29228172"
          >
            link
          </a>
          )
        </ListItem>
        <ListItem>
          version: 
          2.0
           (last updated 
          2018.03.12
          ),
          <a
            href="https://github.com/AllonKleinLab/SPRING_dev"
          >
             github link
          </a>
        </ListItem>
        <ListItem>
          <GridColumn
            floated="right"
          >
            <Button
              as="button"
              basic={true}
              icon={true}
              labelPosition="right"
              role="button"
            >
              <Link
                replace={false}
                to={
                  Object {
                    "pathname": "/dataset",
                    "search": "?viz=spring",
                  }
                }
              >
                launch SPRING
              </Link>
              <Icon
                as="i"
                name="external alternate"
              />
            </Button>
          </GridColumn>
        </ListItem>
      </List>
    </GridColumn>
  </Grid>
  <Accordion
    defaultActiveIndex={0}
    panels={
      Array [
        Object {
          "content": Object {
            "content": <React.Fragment>
              <Divider />
              <Grid
                centered={true}
                columns={1}
                container={true}
                divided="vertically"
                style={
                  Object {
                    "padding": "10px 0",
                  }
                }
              >
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="HPC (hematopoietic progenitor cells) icon"
                        src="assets/icons/example_hpc_sf2-spring.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        HPC (hematopoietic progenitor cells)
                      </Header>
                      <p>
                        Analysis of 33,473 hematopoietic progenitor cells as they differentiate over 6 days.       Barcodes were introduced at day 0 and cell lineage relationships were traced by       identifying progeny from their barcodes in scRNAseq of aliquots of the same population taken at several timepoints.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=hpc_sf2/full&viz=spring"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="CZI Tabula muris icon"
                        src="assets/icons/tabula-muris-thumbnail.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        CZI Tabula muris
                      </Header>
                      <p>
                        Analysis of 10K mouse cells from 20 organs and tissues.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=tabula_muris/10k&viz=spring"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="Tabula Muris - SPRING vs tSNE with Anatomogram icon"
                        src="assets/icons/example_tabularmuris_spring-tsne-anatomogram.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        Tabula Muris - SPRING vs tSNE with Anatomogram
                      </Header>
                      <p>
                        Example comparison of data reduction techniques SPRING and tSNE    (of top 30 PCA components), on RNAseq profiles from 10000 cells    from the CZI Tabula muris dataset. Also includes anatomogram visualization for selection of tissue types.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=tabula_muris/10k&viz=spring&viz=tfjs-tsne&viz=anatomogram"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="HPC - SPRING vs tSNE with Anatomogram icon"
                        src="assets/icons/example_HPC_spring-tsne-anatomogram.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        HPC - SPRING vs tSNE with Anatomogram
                      </Header>
                      <p>
                        Example interaction between SPRING, tSNE and Anatomogram visualization on a small dataset.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=hpc/full&viz=spring&viz=tfjs-tsne&viz=anatomogram"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
              </Grid>
            </React.Fragment>,
          },
          "key": "examples",
          "title": "example stories",
        },
        Object {
          "content": "coming soon!",
          "key": "learn",
          "title": "learn",
        },
      ]
    }
  />
</Container>
`;

exports[`VizOverviewPage Should match existing snapshot for initial visualizations. 3`] = `
<Container>
  <Grid
    centered={true}
    columns={2}
  >
    <GridColumn
      width={2}
    >
      <img
        alt="icon for tfjs-tsne"
        src="assets/icons/tfjs-tsne-icon.png"
        style={
          Object {
            "height": "150px",
            "padding": "20px",
          }
        }
      />
    </GridColumn>
    <GridColumn
      textAlign="left"
    >
      <Header
        as="h1"
      >
        tfjs-tsne
        <HeaderSubheader>
          Yannick Assogba
        </HeaderSubheader>
      </Header>
      <p />
      <List>
        <ListItem>
          applicable data: 
          scRNA-seq
        </ListItem>
        <ListItem>
          compatible with: 
        </ListItem>
        <ListItem>
          citation(s):
           
        </ListItem>
        <ListItem>
          version: 
           (last updated 
          ),
          <a
            href="https://github.com/tensorflow/tfjs-tsne"
          >
             github link
          </a>
        </ListItem>
        <ListItem>
          <GridColumn
            floated="right"
          >
            <Button
              as="button"
              basic={true}
              icon={true}
              labelPosition="right"
              role="button"
            >
              <Link
                replace={false}
                to={
                  Object {
                    "pathname": "/dataset",
                    "search": "?viz=tfjs-tsne",
                  }
                }
              >
                launch tfjs-tsne
              </Link>
              <Icon
                as="i"
                name="external alternate"
              />
            </Button>
          </GridColumn>
        </ListItem>
      </List>
    </GridColumn>
  </Grid>
  <Accordion
    defaultActiveIndex={0}
    panels={
      Array [
        Object {
          "content": Object {
            "content": <React.Fragment>
              <Divider />
              <Grid
                centered={true}
                columns={1}
                container={true}
                divided="vertically"
                style={
                  Object {
                    "padding": "10px 0",
                  }
                }
              >
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="Tabula Muris - SPRING vs tSNE with Anatomogram icon"
                        src="assets/icons/example_tabularmuris_spring-tsne-anatomogram.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        Tabula Muris - SPRING vs tSNE with Anatomogram
                      </Header>
                      <p>
                        Example comparison of data reduction techniques SPRING and tSNE    (of top 30 PCA components), on RNAseq profiles from 10000 cells    from the CZI Tabula muris dataset. Also includes anatomogram visualization for selection of tissue types.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=tabula_muris/10k&viz=spring&viz=tfjs-tsne&viz=anatomogram"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="HPC - SPRING vs tSNE with Anatomogram icon"
                        src="assets/icons/example_HPC_spring-tsne-anatomogram.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        HPC - SPRING vs tSNE with Anatomogram
                      </Header>
                      <p>
                        Example interaction between SPRING, tSNE and Anatomogram visualization on a small dataset.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=hpc/full&viz=spring&viz=tfjs-tsne&viz=anatomogram"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
              </Grid>
            </React.Fragment>,
          },
          "key": "examples",
          "title": "example stories",
        },
        Object {
          "content": "coming soon!",
          "key": "learn",
          "title": "learn",
        },
      ]
    }
  />
</Container>
`;

exports[`VizOverviewPage Should match existing snapshot when changing the visualization. 1`] = `
<Container>
  <Grid
    centered={true}
    columns={2}
  >
    <GridColumn
      width={2}
    >
      <img
        alt="icon for Anatomogram"
        src="assets/icons/anatomogram-icon.png"
        style={
          Object {
            "height": "150px",
            "padding": "20px",
          }
        }
      />
    </GridColumn>
    <GridColumn
      textAlign="left"
    >
      <Header
        as="h1"
      >
        Anatomogram
        <HeaderSubheader>
          Alfonso Mu√±oz-Pomer Fuentes
        </HeaderSubheader>
      </Header>
      <p />
      <List>
        <ListItem>
          applicable data: 
          scRNA-seq
        </ListItem>
        <ListItem>
          compatible with: 
        </ListItem>
        <ListItem>
          citation(s):
           
        </ListItem>
        <ListItem>
          version: 
           (last updated 
          ),
          <a
            href="https://github.com/ebi-gene-expression-group/anatomogram"
          >
             github link
          </a>
        </ListItem>
        <ListItem>
          <GridColumn
            floated="right"
          >
            <Button
              as="button"
              basic={true}
              icon={true}
              labelPosition="right"
              role="button"
            >
              <Link
                replace={false}
                to={
                  Object {
                    "pathname": "/dataset",
                    "search": "?viz=anatomogram",
                  }
                }
              >
                launch Anatomogram
              </Link>
              <Icon
                as="i"
                name="external alternate"
              />
            </Button>
          </GridColumn>
        </ListItem>
      </List>
    </GridColumn>
  </Grid>
  <Accordion
    defaultActiveIndex={0}
    panels={
      Array [
        Object {
          "content": Object {
            "content": <React.Fragment>
              <Divider />
              <Grid
                centered={true}
                columns={1}
                container={true}
                divided="vertically"
                style={
                  Object {
                    "padding": "10px 0",
                  }
                }
              >
                <GridRow
                  columns={2}
                >
                  <React.Fragment>
                    <GridColumn
                      width={2}
                    >
                      <img
                        alt="Tabula Muris - SPRING vs tSNE with Anatomogram icon"
                        src="assets/icons/example_tabularmuris_spring-tsne-anatomogram.png"
                        style={
                          Object {
                            "height": "75px",
                            "width": "75px",
                          }
                        }
                      />
                    </GridColumn>
                    <GridColumn
                      textAlign="left"
                      width={8}
                    >
                      <Header>
                        Tabula Muris - SPRING vs tSNE with Anatomogram
                      </Header>
                      <p>
                        Example comparison of data reduction techniques SPRING and tSNE    (of top 30 PCA components), on RNAseq profiles from 10000 cells    from the CZI Tabula muris dataset. Also includes anatomogram visualization for selection of tissue types.
                      </p>
                    </GridColumn>
                    <GridColumn
                      floated="right"
                    >
                      <Button
                        as="button"
                        basic={true}
                        icon={true}
                        labelPosition="right"
                        role="button"
                      >
                        <Link
                          replace={false}
                          to="/dataset?name=tabula_muris/10k&viz=spring&viz=tfjs-tsne&viz=anatomogram"
                        >
                          launch example
                        </Link>
                        <Icon
                          as="i"
                          name="external alternate"
                        />
                      </Button>
                    </GridColumn>
                  </React.Fragment>
                </GridRow>
              </Grid>
            </React.Fragment>,
          },
          "key": "examples",
          "title": "example stories",
        },
        Object {
          "content": "coming soon!",
          "key": "learn",
          "title": "learn",
        },
      ]
    }
  />
</Container>
`;

exports[`VizOverviewPage Should match existing snapshot when no props are provided. 1`] = `""`;
